### ISSUE TITLE
Build Error

### ISSUE THREAD

**Issue by junjihashimoto** [nc6/inline-c-test#1]
Hi.

I can not build this with ghc7.10 and incline-c-0.5.4.
What's wrong?

```
Building inline-c-test-0.1.0.0...
Preprocessing library inline-c-test-0.1.0.0...
[1 of 2] Compiling Context          ( dist/dist-sandbox-887793ee/build/Context.hs, dist/dist-sandbox-887793ee/build/Context.o )
[2 of 2] Compiling Internal         ( dist/dist-sandbox-887793ee/build/Internal.hs, dist/dist-sandbox-887793ee/build/Internal.o )

src/Internal.hsc:21:15:
    Exception when trying to run compile-time code:
      CIdentifier fromString: invalid string "struct a0_foo"
"cIdentifierFromString" (line 1, column 7):
unexpected reserved C identifier "struct"
    Code: template-haskell-2.10.0.0:Language.Haskell.TH.Quote.quoteExp
            C.block
            " int {\n\
            \    struct a0_foo *arr = $(struct a0_foo *arr);\n\
            \    free(arr);\n\
            \  }\n"
cabal: Error: some packages failed to install:
inline-c-test-0.1.0.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by nc6** [comment]
I created this repo as an example for a bug report in `inline-c`, so it's not meant to compile! Actually, the issue turns out to be with the context - see https://github.com/fpco/inline-c/issues/25

If you're interested in using this repo for anything else, I'll push the changes which get it building.

**Comment by nc6** [comment]
Should be fixed by 51d260b9845dfe95af47c123b1472e3543354a7c

**Comment by junjihashimoto** [comment]
Thank you for your great help.
